awx/ui/src/screens/TopologyView/MeshGraph.js

```diff
@@ -0,0 +1,274 @@
+  width: 100%;
+  background: white;
+`;
+function MeshGraph({ data, showLegend, zoom }) {
+  // function MeshGraph({ showLegend, zoom }) {
+  const [isNodeSelected, setIsNodeSelected] = useState(false);
+  const [selectedNode, setSelectedNode] = useState(null);
+  const [nodeDetail, setNodeDetail] = useState(null);
+  const history = useHistory();
+
+  // const data = generateRandomNodes(getRandomInt(4, 50));
+  const draw = () => {
+    const width = getWidth(SELECTOR);
+    const height = getHeight(SELECTOR);
+
+    /* Add SVG */
+    d3.selectAll(`#chart > svg`).remove();
+    const svg = d3
+      .select('#chart')
+      .append('svg')
+      .attr('class', 'mesh-svg')
+      .attr('width', `${width}px`)
+      .attr('height', `100%`);
+    const mesh = svg.append('g').attr('class', 'mesh');
+
+    const graph = data;
+
+    const simulation = d3
+      .forceSimulation(graph.nodes)
+      .force(
+        'charge',
+        d3
+          .forceManyBody(MESH_FORCE_LAYOUT.defaultForceBody)
+          .strength(MESH_FORCE_LAYOUT.defaultForceStrength)
+      )
+      .force(
+        'link',
+        d3.forceLink(graph.links).id((d) => d.hostname)
+      )
+      .force(
+        'collide',
+        d3.forceCollide(MESH_FORCE_LAYOUT.defaultCollisionFactor)
+      )
+      .force('forceX', d3.forceX(MESH_FORCE_LAYOUT.defaultForceX))
+      .force('forceY', d3.forceY(MESH_FORCE_LAYOUT.defaultForceY))
+      .force('center', d3.forceCenter(width / 2, height / 2));
+
+    const link = mesh
+      .append('g')
+      .attr('class', `links`)
+      .attr('data-cy', 'links')
+      .selectAll('line')
+      .data(graph.links)
+      .enter()
+      .append('line')
+      .attr('class', (_, i) => `link-${i}`)
+      .attr('data-cy', (d) => `${d.source.hostname}-${d.target.hostname}`)
+      .style('fill', 'none')
+      .style('stroke', '#ccc')
+      .style('stroke-width', '2px')
+      .attr('pointer-events', 'none')
+      .on('mouseover', function showPointer() {
+        d3.select(this).transition().style('cursor', 'pointer');
+      });
+
+    const node = mesh
+      .append('g')
+      .attr('class', 'nodes')
+      .attr('data-cy', 'nodes')
+      .selectAll('g')
+      .data(graph.nodes)
+      .enter()
+      .append('g')
+      .on('mouseenter', function handleNodeHover(_, d) {
+        d3.select(this).transition().style('cursor', 'pointer');
+        highlightSiblings(d);
+      })
+      .on('mouseleave', (_, d) => {
+        deselectSiblings(d);
+      })
+      .on('click', (_, d) => {
+        setNodeDetail(d);
+        highlightSelected(d);
+      });
+
+    // node circles
+    node
+      .append('circle')
+      .attr('r', DEFAULT_RADIUS)
+      .attr('class', (d) => d.node_type)
+      .attr('class', (d) => `id-${d.id}`)
+      .attr('fill', DEFAULT_NODE_COLOR)
+      .attr('stroke', DEFAULT_NODE_LABEL_TEXT_COLOR);
+
+    // node type labels
+    node
+      .append('text')
+      .text((d) => renderNodeType(d.node_type))
```

Comment:
Would possible to place the text at center of the node?

![image](https://user-images.githubusercontent.com/9053044/154985525-5a368e17-fafd-4e3d-8612-cb15e2d0a72e.png)

Reply:
This is happening on Firefox browser

Reply:
This should now be resolved with the latest [commit](https://github.com/ansible/awx/pull/11501/commits/fc2f6f632567cffe67fd0b9ba5ac06d59b0e31b7).